src/datachain/query/session.py

```diff
@@ -114,9 +139,23 @@ def get(
                 in_memory=in_memory,
             )
             cls.GLOBAL_SESSION = cls.GLOBAL_SESSION_CTX.__enter__()
+
             atexit.register(cls._global_cleanup)
+            cls.ORIGINAL_EXCEPT_HOOK = sys.excepthook
+            sys.excepthook = cls.except_hook
```

Comment:
why does `atexit` not enough?

is it needed for Studio?

Reply:
atexit doesn't work when unknown exceptions are raised.

Reply:
what is an "unknown exception"? I see the code above:

```
        if exc_type:
            self._cleanup_created_versions(self.name)
```

that looks like it does the same? no?

Reply:
This is when the session is used as a context manager only.

Reply:
okay, I see ... it was not obvious that we want to cleanup on a non-global context managed sessions only (vs let users deal with it, e.g. if some exceptions are expected and handled), but it makes sense I guess ... let's keep it!